packages/vue-components/src/components/OmegaForm/OmegaWrapper.vue

```diff
@@ -112,22 +112,26 @@ const localForm = props.form || !props.schema
     {
       ...props,
       onSubmit: typeof props.isLoading !== "undefined"
-        ? ({ value }) =>
-          new Promise<void>((resolve) => {
-            console.log("Submitting form with values:", value, "emitting")
-            instance!.emit("submit", value)
-            // even if the emit would be immediately handled, prop changes are not published/received immediately.
-            const handle = watch(() => props.isLoading, (v) => {
-              if (v) return
-              resolve()
-              handle.stop()
-            })
-          })
+        ? undefined
         : props.onSubmit
     },
     props.omegaConfig
   )
 
+const instance = getCurrentInstance()
+
+const handleFormSubmit = (): void => {
+  if (formIsSubmitting.value) return
+
+  formToUse.value.handleSubmit().then(() => {
```

Comment:
handleSubmit internally calls onSubmit or onSubmitInvalid, I know you check the state after, but imo we're getting into messy territory :)

Reply:
I agree, but it's still less scarier than debug a watch inside a promise resolution IMHO